app/controllers/lectures_controller.rb

```diff
@@ -66,6 +67,7 @@ def edit
 
   def create
     @lecture = Lecture.new(lecture_params)
+    @lecture.teacher = current_user unless current_user.admin?
```

Comment:
To understand this line, am I right that:
- `@lecture.teacher = current_user` was done beforehand implicitly by Devise
- `unless current_user.admin?` is needed such that upon creation of a new lecture, the teacher field is empty.

Reply:
Beforehand, the line

```ruby
allowed_params.push(:course_id, :teacher_id, { editor_ids: [] }) if action_name == "create"
```

in the `lecture_params` method made sure that `teacher_id` is whitelisted, and then

```
@lecture = Lecture.new(lecture_params)
```
would make sure that `@lecture.teacher_id` is set as in the parameter hash. Of course, the create action's authorization is checked in the corresponding ability file, so anyone who passes authorization (i.e. course editors or admins) could make an arbitrary user the teacher of the new lecture. Now the line

```ruby
allowed_params.push(:teacher_id) if current_user.admin?
```
makes sure that `teacher_id` is whitelisted only for admins. But in order to get a valid lecture, the `teacher_id`
needs to be set also in case the lecture is ccreated by a course editor that is not an admin. But the new default here is that users can only set themselves as teachers. This should also be reflected in the corresponding forms for creating a lecture.